spacy/pipeline/trainable_pipe.pyx

```diff
@@ -56,6 +56,56 @@ cdef class TrainablePipe(Pipe):
         except Exception as e:
             error_handler(self.name, self, [doc], e)
 
+
+    def distill(self,
+               teacher_pipe: Optional["TrainablePipe"],
+               teacher_docs: Iterable["Doc"],
+               student_docs: Iterable["Doc"],
```

Comment:
In a sense, the `teacher_docs` and `student_docs` do form a batch of `Example` objects, no? It's just that the "reference" in the example isn't exactly "gold data", it's "teacher predictions". Still, it might all be more straightforward if we pass in a batch of `Example` objects instead? It'd also make the distill loss function look much more like a "normal" loss function - if not in fact entirely the same?

Reply:
Hmmm, it would be a bit weird though, since you'd expect `Example.reference` to have annotations, but we don't let the teacher annotate the `Doc` in the usual case (currently only for `tok2vec`). But does clean up things quite a bit.